qiskit_ibm_runtime/api/rest/cloud_backend.py

```diff
@@ -0,0 +1,99 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""IBM Cloud Backend REST adapter."""
+
+from typing import Dict, Any
+
+from .base import RestAdapterBase
+from ..session import RetrySession
+
+
+class CloudBackend(RestAdapterBase):
+    """Rest adapter for IBM Cloud backend related endpoints."""
+
+    URL_MAP = {
+        "configuration": "/configuration",
+        "properties": "/properties",
+        "pulse_defaults": "/defaults",
+        "status": "/status",
+    }
+
+    def __init__(
+        self, session: RetrySession, backend_name: str, url_prefix: str = ""
+    ) -> None:
+        """Backend constructor.
+
+        Args:
+            session: Session to be used in the adaptor.
+            backend_name: Name of the backend.
+            url_prefix: Base URL.
+        """
+        self.backend_name = backend_name
+        super().__init__(session, "{}/devices/{}".format(url_prefix, backend_name))
+
+    def configuration(self) -> Dict[str, Any]:
+        """Return backend configuration.
+
+        Returns:
+            JSON response of backend configuration.
+        """
+        url = self.get_url("configuration")
+        return self.session.get(url).json()
+
+    def properties(self) -> Dict[str, Any]:
+        """Return backend properties.
+
+        Returns:
+            JSON response of backend properties.
+        """
+        # pylint: disable=redefined-outer-name
+        url = self.get_url("properties")
+        response = self.session.get(url).json()
+        # Adjust name of the backend.
+        if response:
+            response["backend_name"] = self.backend_name
+        return response
+
+    def pulse_defaults(self) -> Dict[str, Any]:
+        """Return backend pulse defaults.
+
+        Returns:
+            JSON response of pulse defaults.
+        """
+        url = self.get_url("pulse_defaults")
+        return self.session.get(url).json()
+
+    def status(self) -> Dict[str, Any]:
+        """Return backend status.
+
+        Returns:
+            JSON response of backend status.
+        """
+        url = self.get_url("status")
+        response = self.session.get(url).json()
+        # Adjust fields according to the specs (BackendStatus).
+        ret = {
+            "backend_name": self.backend_name,
+            "backend_version": response.get("backend_version", "0.0.0"),
+            "status_msg": response.get("status", ""),
+            "operational": bool(response.get("state", False)),
+        }
+        # 'pending_jobs' is required, and should be >= 0.
+        if "length_queue" in response:
+            ret["pending_jobs"] = max(response["length_queue"], 0)
+        else:
+            ret["pending_jobs"] = 0
+        # Not part of the schema.
+        if "busy" in response:
+            ret["dedicated"] = response["busy"]
+        return ret
```

Comment:
This is mostly a duplicate of https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/qiskit_ibm_runtime/api/rest/backend.py#L85-L112. Just a little detail appears to differ (`length_queue` vs. `lengthQueue`).

Do we expect both code paths to evolve independently from here on or should we strive for more reuse?

Reply:
For the reasons mentioned in this comment https://github.com/Qiskit/qiskit-ibm-runtime/pull/46#discussion_r763095193, I have this function copied in a separate file. If we decide to put it all in one file then it'll address above comment. Else we can discuss how to reuse. But since they connect to two different API services it would be nice to keep these separate.

Reply:
Convinced it makes sense to keep separate files. Maybe you find a simple way to delegate calls from `cloud_backend.py#status` to `backend.py#status` to avoid duplicating many lines as long as the APIs are very similar.

Reply:
I don't want to combine the two paths, but yes I promise to get back to this once we get everything working.